pydantic/main.py

```diff
@@ -810,6 +564,223 @@ def model_parametrized_name(cls, params: tuple[type[Any], ...]) -> str:
+        exclude_defaults: bool = False,
+        exclude_none: bool = False,
+        # TODO: What do we do about the following arguments?
+        #   Do they need to go on model_config now, and get used by the serializer?
+        encoder: typing.Callable[[Any], Any] | None = None,
+        models_as_dict: bool = True,
+        **dumps_kwargs: Any,
+    ) -> str:
+        warnings.warn('The `json` method is deprecated; use `model_dump_json` instead.', DeprecationWarning)
+        return self.model_dump_json(
+            include=include,
+            exclude=exclude,
+            by_alias=by_alias,
+            exclude_unset=exclude_unset,
+            exclude_defaults=exclude_defaults,
+            exclude_none=exclude_none,
+        )
+
+    @classmethod
+    def parse_obj(cls: type[Model], obj: Any) -> Model:
+        warnings.warn('The `parse_obj` method is deprecated; use `model_validate` instead.', DeprecationWarning)
+        return cls.model_validate(obj)
+
+    @classmethod
+    def parse_raw(
+        cls: type[Model],
+        b: str | bytes,
+        *,
+        content_type: str = None,
+        encoding: str = 'utf8',
+        proto: _deprecated_parse.Protocol = None,
+        allow_pickle: bool = False,
+    ) -> Model:
+        warnings.warn(
+            'The `parse_raw` method is deprecated; load the data and use `model_validate` instead.', DeprecationWarning
+        )
+        try:
+            obj = _deprecated_parse.load_str_bytes(
+                b,
+                proto=proto,
+                content_type=content_type,
+                encoding=encoding,
+                allow_pickle=allow_pickle,
+                json_loads=cls.model_config['json_loads'],
+            )
+        except (ValueError, TypeError, UnicodeDecodeError) as e:
+            # TODO: this used to raise ValidationError([ErrorWrapper(e, loc=ROOT_KEY)], cls)
+            #   Do we want to make it possible to instantiate from python, just for deprecated functionality?
+            raise e
+        return cls.parse_obj(obj)
+
+    @classmethod
+    def parse_file(
+        cls: type[Model],
+        path: str | Path,
+        *,
+        content_type: str = None,
+        encoding: str = 'utf8',
+        proto: _deprecated_parse.Protocol = None,
+        allow_pickle: bool = False,
+    ) -> Model:
+        warnings.warn(
+            'The `parse_file` method is deprecated; load the data and use `model_validate` instead.', DeprecationWarning
+        )
+        obj = _deprecated_parse.load_file(
+            path,
+            proto=proto,
+            content_type=content_type,
+            encoding=encoding,
+            allow_pickle=allow_pickle,
+            json_loads=cls.model_config['json_loads'],
+        )
+        return cls.parse_obj(obj)
+
+    @classmethod
+    def from_orm(cls: type[Model], obj: Any) -> Model:
+        warnings.warn(
+            'The `from_orm` method is deprecated; set model_config["from_attributes"]=True '
```

Comment:
```suggestion
            'The `from_orm` method is deprecated; set `model_config["from_attributes"]=True` '
```

Reply:
I guess we can remove the note about `model_config["from_attributes"]=True` if they already have it set and we're not raising an error.

Reply:
on reflection, I think this is fine.